src/Components/Components/src/PersistingComponentStateSubscription.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Infrastructure;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Represents a subscription to the OnPersisting callback that <see cref="ComponentStatePersistenceManager"/> will trigger
+    /// when the application is being persisted.
+    /// </summary>
+    public struct PersistingComponentStateSubscription : IDisposable
+    {
+        private readonly List<Func<Task>> _callbacks;
```

Comment:
For structs, it's safer to annotate all fields as nullable to account for the default ctor.